include/openenclave/host.h

```diff
@@ -72,7 +72,7 @@ OE_EXTERNC_BEGIN
 /**
  *  Flag passed into oe_create_enclave to run the enclave in simulation mode.
  */
-#define OE_ENCLAVE_FLAG_SIMULATE 0x00000002u
+#define OE_ENCLAVE_FLAG_SIMULATE 0x00000020u
```

Comment:
Can you explain on this change?

Reply:
When we use oesign and set debug=1, the properties.config.attribute gets set to |= SGX_FLAGS_DEBUG - https://github.com/openenclave/openenclave/blob/f8b79cf8eb64874ccd0cae951ff7f52672f1d126/tools/oesign/oesign.c#L518-L524.Later when the sgx_build_enclave() is called in host/sgx/create.c, the enclave->debug is set to 0 and enclave->simulate are both being set to 1. This is because we check to see if the value attributes has bit OE_ENCLAVE_FLAG_DEBUG set for debug and OE_ENCLAVE_FLAG_SIMULATION for simulate  - https://github.com/openenclave/openenclave/blob/f8b79cf8eb64874ccd0cae951ff7f52672f1d126/host/sgx/create.c#L827-L828.This is because OE_ENCLAVE_FLAG_SIMULATION = 0x2 and so is SGX_FLAGS_DEBUG = 0x2.

The problem was that when we created enclaves in debug mode, using oesign, they were actually being created in simulation mode.

Reply:
Added a function to map attributes in oesign.c from SGX_FLAGS_* to OE_ENCLAVE_FLAG_*.